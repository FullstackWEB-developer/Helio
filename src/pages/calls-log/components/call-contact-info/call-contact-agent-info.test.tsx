import i18n from '../../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import CallContactAgentInfo from './call-contact-agent-info';

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("Call Contact Agent Info tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                
            }
        },
        lookupsState: {
            userList: [{
                id: "123",
                firstName: "Emre",
                lastName: "Hanci",
            }]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        callsLogState: {
            isFiltered: false
        },
        ticketState: {
            lookupValues: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders call-contact-agent-info correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
                <CallContactAgentInfo agentId={"123"} type={'CHAT'}/>           
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders call-contact-agent-info-without-user correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
                <CallContactAgentInfo agentId={"1234"} type={'CHAT'}/>           
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders call-contact-agent-info-without-agent correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
                <CallContactAgentInfo type={'CHAT'}/>           
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders call-contact-agent-info-without-agent-voice correctly", async () => {
        const {asFragment} = render(<TestWrapper mockState={mockState}>
                <CallContactAgentInfo type={'VOICE'}/>           
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders call-contact-agent-info-without-user-name correctly", async () => {
        const mockState = {
            emailState: {
                unreadEmails: 0,
                messageSummaries: []
            },
            appUserState: {
                liveAgentStatuses: [],
                appUserDetails: {
                    
                }
            },
            lookupsState: {
                userList: [{
                    id: "123"
                }]
            },
            appState: {
                smsTemplates: [],
                emailTemplates: []
            },
            callsLogState: {
                isFiltered: false
            },
            ticketState: {
                lookupValues: []
            }
        };
        const {asFragment} = render(<TestWrapper mockState={mockState}>
                <CallContactAgentInfo agentId={"123"} type={'CHAT'}/>           
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
