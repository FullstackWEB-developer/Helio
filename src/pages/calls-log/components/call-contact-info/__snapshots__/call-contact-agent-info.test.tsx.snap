// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Call Contact Agent Info tests renders call-contact-agent-info correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="avatar rounded-full flex items-center justify-center relative w-10 h-10"
    >
      <div
        class="avatar-initial"
      >
        EH
      </div>
    </div>
    <span
      class="ml-4 body2"
    >
      Emre Hanci
    </span>
  </div>
</DocumentFragment>
`;

exports[`Call Contact Agent Info tests renders call-contact-agent-info-without-agent correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="avatar rounded-full flex items-center justify-center relative w-10 h-10"
    >
      <div
        class=""
      >
        <svg
          class="icon-medium "
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,0H24V24H0Z"
            fill="none"
          />
          <path
            class="white-icon"
            d="M5.333,15a.836.836,0,0,0,.833.833H7V18.75a1.25,1.25,0,1,0,2.5,0V15.833h1.667V18.75a1.25,1.25,0,0,0,2.5,0V15.833H14.5A.836.836,0,0,0,15.333,15V6.667h-10ZM3.25,6.667A1.248,1.248,0,0,0,2,7.917V13.75a1.25,1.25,0,1,0,2.5,0V7.917A1.248,1.248,0,0,0,3.25,6.667Zm14.167,0a1.248,1.248,0,0,0-1.25,1.25V13.75a1.25,1.25,0,0,0,2.5,0V7.917A1.248,1.248,0,0,0,17.417,6.667ZM13.275,1.8,14.358.717a.413.413,0,0,0,0-.592.413.413,0,0,0-.592,0L12.533,1.358a4.866,4.866,0,0,0-2.2-.525,4.936,4.936,0,0,0-2.217.525L6.875.125a.413.413,0,0,0-.592,0,.413.413,0,0,0,0,.592L7.375,1.808A4.986,4.986,0,0,0,5.333,5.833h10A4.971,4.971,0,0,0,13.275,1.8ZM8.667,4.167H7.833V3.333h.833Zm4.167,0H12V3.333h.833Z"
            transform="translate(2 2)"
          />
        </svg>
      </div>
    </div>
    <span
      class="ml-4 body2"
    >
      common.chat common.bot
    </span>
  </div>
</DocumentFragment>
`;

exports[`Call Contact Agent Info tests renders call-contact-agent-info-without-agent-voice correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="avatar rounded-full flex items-center justify-center relative w-10 h-10"
    >
      <div
        class=""
      >
        <svg
          class="icon-medium "
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,0H24V24H0Z"
            fill="none"
          />
          <path
            class="white-icon"
            d="M5.333,15a.836.836,0,0,0,.833.833H7V18.75a1.25,1.25,0,1,0,2.5,0V15.833h1.667V18.75a1.25,1.25,0,0,0,2.5,0V15.833H14.5A.836.836,0,0,0,15.333,15V6.667h-10ZM3.25,6.667A1.248,1.248,0,0,0,2,7.917V13.75a1.25,1.25,0,1,0,2.5,0V7.917A1.248,1.248,0,0,0,3.25,6.667Zm14.167,0a1.248,1.248,0,0,0-1.25,1.25V13.75a1.25,1.25,0,0,0,2.5,0V7.917A1.248,1.248,0,0,0,17.417,6.667ZM13.275,1.8,14.358.717a.413.413,0,0,0,0-.592.413.413,0,0,0-.592,0L12.533,1.358a4.866,4.866,0,0,0-2.2-.525,4.936,4.936,0,0,0-2.217.525L6.875.125a.413.413,0,0,0-.592,0,.413.413,0,0,0,0,.592L7.375,1.808A4.986,4.986,0,0,0,5.333,5.833h10A4.971,4.971,0,0,0,13.275,1.8ZM8.667,4.167H7.833V3.333h.833Zm4.167,0H12V3.333h.833Z"
            transform="translate(2 2)"
          />
        </svg>
      </div>
    </div>
    <span
      class="ml-4 body2"
    >
      common.voice common.bot
    </span>
  </div>
</DocumentFragment>
`;

exports[`Call Contact Agent Info tests renders call-contact-agent-info-without-user correctly 1`] = `<DocumentFragment />`;

exports[`Call Contact Agent Info tests renders call-contact-agent-info-without-user-name correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="avatar rounded-full flex items-center justify-center relative w-10 h-10"
    >
      <div
        class="avatar-initial"
      />
    </div>
    <span
      class="ml-4 body2"
    >
       
    </span>
  </div>
</DocumentFragment>
`;
